controllers/chat/moveToAnon.js

```diff
@@ -77,6 +99,13 @@ const moveToAnon = async (req, res) => {
       newStateMemberWithAnonInfo[prevTargetUser.user_id].anon_user_info_emoji_name =
         prevTargetUser?.anon_user_info_emoji_name;
 
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].user = {};
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].user.id = prevTargetUser.user_id;
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].user.name =
+        'Anonymous ' + prevTargetUser?.anon_user_info_emoji_name;
+      newStateMemberWithAnonInfo[prevTargetUser.user_id].user.username =
+        'Anonymous ' + prevTargetUser?.anon_user_info_emoji_name;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `newStateMemberWithAnonInfo` object is updated with anonymous user information and a new `user` object is created with the anonymous user's ID, name, and username. This seems to be a reasonable change, but ensure that this new logic aligns with your application's requirements.
```diff
+       newStateMemberWithAnonInfo[prevTargetUser.user_id].user = {};
+       newStateMemberWithAnonInfo[prevTargetUser.user_id].user.id = prevTargetUser.user_id;
+       newStateMemberWithAnonInfo[prevTargetUser.user_id].user.name =
+         'Anonymous ' + prevTargetUser?.anon_user_info_emoji_name;
+       newStateMemberWithAnonInfo[prevTargetUser.user_id].user.username =
+         'Anonymous ' + prevTargetUser?.anon_user_info_emoji_name;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->